src/contexts/html.ts

```diff
@@ -0,0 +1,143 @@
+/**
+ * Copyright 2017-present Palantir Technologies, Inc. All rights reserved.
+ * Licensed under the MIT License (the "License"); you may obtain a copy of the
+ * license at https://github.com/palantir/typesettable/blob/develop/LICENSE
+ */
+
+import { IDimensions } from "../measurers";
+import { IAnchor, ITransform } from "../writers";
+import { ITypesetterContext } from "./index";
+
+export class HtmlUtils {
+  /**
+   * Appends an SVG element with the specified tag name to the provided element.
+   * The variadic classnames are added to the new element.
+   *
+   * Returns the new element.
+   */
+  public static append(element: Element, tagName: string, ...classNames: string[]) {
+    const child = HtmlUtils.create(tagName, ...classNames);
+    element.appendChild(child);
+    return child;
+  }
+
+  /**
+   * Creates and returns a new SVGElement with the attached classnames.
+   */
+  public static create(tagName: string, ...classNames: string[]) {
+    const element = document.createElement(tagName) as HTMLElement;
+    HtmlUtils.addClasses(element, ...classNames);
+    return element;
+  }
+
+  /**
+   * Adds the variadic classnames to the element
+   */
+  public static addClasses(element: Element, ...classNames: string[]) {
+    classNames = classNames.filter((c) => c != null);
+    if (element.classList != null) {
+      classNames.forEach((className) => {
+        element.classList.add(className);
+      });
+    } else {
+      // IE 11 does not support classList
+      element.setAttribute("class", classNames.join(" "));
+    }
+  }
+
+  /**
+   * Returns the width/height of svg element's bounding box
+   */
+  public static getDimensions(element: HTMLElement): IDimensions {
+    // using feature detection, safely return the bounding box dimensions of the
+    // provided svg element
+    if (element.getBoundingClientRect) {
```

Comment:
we shouldn't need feature detection on `getBoundingClientRect`; it's definitely supported in all the stuff we care about